src/Api/Serializer/RestSerializer.php

```diff
@@ -126,6 +126,13 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
         if ($member->getType() == 'timestamp') {
             $value = TimestampShape::format($value, 'rfc822');
         }
+        if ($member['jsonvalue']) {
+            $value = json_encode($value);
+            if ($value === null && json_last_error() !== JSON_ERROR_NONE) {
+                throw new \InvalidArgumentException('Unable to encode the provided value as JSON');
```

Comment:
[PayloadParserTrait](https://github.com/aws/aws-sdk-php/blob/master/src/Api/Parser/PayloadParserTrait.php) has methods only for decoding JSON and not for encoding.